### ISSUE TITLE
About Evaluation metrics for BraTS example

### ISSUE THREAD

**Issue by tarekegn82** [ellisdg/3DUnetCNN#316]
Dear Ellis,

Thank you for your earlier responses, I have managed to run fivefold cross-validation training on the BraTS dataset. As you know the dice metric used here is the loss function. Similarly, can we generate a `dice coefficient score and HD` of each tumor class `(WT, ET, and TC)`?

Best.

**Comment by ellisdg** [comment]
Hi @tarekegn82, a simple way to generate Dice and HD is to use Monai. I think something along the lines of the following should work:
```
from monai.transforms import LoadImage
from monai.metrics import DiceMetric, HausdorffDistanceMetric
import torch

loader = LoadImage(image_only=True, ensure_channel_first=True)

pred_filename = "/path/to/filename.nii.gz"
label_filename = "/path/to/label_filename.nii.gz"

pred = loader(pred_filename)
label = loader(label_filename)

# transform prediction to labels
pred_labels = pred.argmax(dim=-4)

# TODO: get a separate mask for the WT, ET, TC
# pred_mask = ?

dice = DiceMetric()(pred_mask, label)
hd = HausdorffDistanceMetric()(pred_mask, label)
```

My plan is to write a function to do this automatically when cross validation is run, but I haven't found the time to do it yet. But if you or someone else is interested in writing the code to have it run during cross validation, that would be awesome!

**Comment by tarekegn82** [comment]
Thank you for the tip, I will try it this way and see if it works. I will also try to write the code into the cross-validation pipeline. In the meantime, please let me know if you do the coding for the latter.
